legal-api/tests/unit/services/filings/validations/test_restoration.py

```diff
@@ -201,26 +201,30 @@ def test_restoration_court_orders(session, test_status, file_number, expected_co
+    'test_status, filing_sub_type, legal_types, nr_number, nr_type, new_legal_name, expected_code, expected_msg',
     [
         # full restoration
-        ('SUCCESS_NEW_NR', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', 'new name', None, None),
-        ('SUCCESS_NAME_ONLY', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], None, 'new name', None, None),
-        ('FAIL_NO_NR_AND_NAME', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], None, None, HTTPStatus.BAD_REQUEST,
+        ('SUCCESS_NEW_NR', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', 'RCC','new name', None, None),
+        ('SUCCESS_NAME_ONLY', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], None, 'RCR', 'new name', None, None),
+        ('FAIL_NO_NR_AND_NAME', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], None, 'BERE', None, HTTPStatus.BAD_REQUEST,
          'Legal name is missing in nameRequest.'),
-        ('FAIL_NR_AND_NO_NAME', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', None,
+        ('FAIL_NR_AND_NO_NAME', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', 'RUL', None,
          HTTPStatus.BAD_REQUEST, 'Legal name is missing in nameRequest.'),
+        ('FAIL_NO_NR_TYPE', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', '', 'new name',
+         HTTPStatus.BAD_REQUEST, 'The name type associated with the name request number entered cannot be used.'),
 
         # limited restoration
-        ('SUCCESS_NEW_NR', 'limitedRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', 'new name', None, None),
-        ('SUCCESS_NAME_ONLY', 'limitedRestoration', ['BC', 'BEN', 'ULC', 'CC'], None, 'new name', None, None),
-        ('FAIL_NO_NR_AND_NAME', 'limitedRestoration', ['BC', 'BEN', 'ULC', 'CC'], None, None, HTTPStatus.BAD_REQUEST,
+        ('SUCCESS_NEW_NR', 'limitedRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', 'RCC', 'new name', None, None),
+        ('SUCCESS_NAME_ONLY', 'limitedRestoration', ['BC', 'BEN', 'ULC', 'CC'], None, 'RCR', 'new name', None, None),
+        ('FAIL_NO_NR_AND_NAME', 'limitedRestoration', ['BC', 'BEN', 'ULC', 'CC'], None, 'BERE', None, HTTPStatus.BAD_REQUEST,
          'Legal name is missing in nameRequest.'),
-        ('FAIL_NR_AND_NO_NAME', 'limitedRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', None,
+        ('FAIL_NR_AND_NO_NAME', 'limitedRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', 'RUL', None,
          HTTPStatus.BAD_REQUEST, 'Legal name is missing in nameRequest.'),
+        ('FAIL_NO_NR_TYPE', 'limitedRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', '', 'new name',
+         HTTPStatus.BAD_REQUEST, 'The name type associated with the name request number entered cannot be used.'),
     ]
 )
-def test_restoration_nr(session, mocker, test_status, filing_sub_type, legal_types, nr_number, new_legal_name,
+def test_restoration_nr(session, mocker, test_status, filing_sub_type, legal_types, nr_number, nr_type, new_legal_name,
```

Comment:
I think we should create a new test to test nr request type specifically.  It's probably possible to include it in this test but it's getting a little messy and we'll need to add some special cases.

I was envisioning the new test to look similar to this but with an additional for loop to loop through an array of nr request types.

With the current test, I would revert the code for this test and maybe update `nr_response` at the top of this `test_restoration.py` to have a valid restoration `requestTypeCd` value.

Reply:
I was thinking something like this for the new test.  You'll of course need to look at `limitedRestoration` as well.

```
('SUCCESS_NR', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', ['RCC', 'RCR', 'BERE', 'RUL'], None, None),
('FAIL_NR', 'fullRestoration', ['BC', 'BEN', 'ULC', 'CC'], 'NR 1234567', ['RCC', 'RCR', 'BERE', 'RUL'], None, None),
def test_restoration_nr_type(session, mocker, test_status, filing_sub_type, legal_types, nr_number, nr_type, expected_code, expected_msg):
```

Reply:
I updated